iio/iio_app/iio_app.h

```diff
@@ -71,21 +74,54 @@ struct iio_app_device {
 };
 
 /**
- * @brief Register devices and start an iio application
- *
- * Configuration for communication is done in parameters.h
- * @param ctx_attrs - array of context attribute name/value pairs.
- * @param nb_ctx_attr - number of context attributes in the array above.
- * @param devices - is an array of devices to register to iiod
- * @param len - is the number of devices
- * @return 0 on success, negative value otherwise
+ * @struct iio_app_desc
+ * @brief IIO application descriptor.
+ */
+struct iio_app_desc {
+	/** Array of context attribute name/value pairs */
+	struct iio_ctx_attr *ctx_attrs;
+	/** Number of context attributes in the array above */
+	uint32_t nb_ctx_attr;
+	struct iio_app_device *devices;
+	uint32_t nb_devices;
+	struct no_os_uart_init_param *user_uart_params;
+	/** IIO triggers to be used */
+	struct iio_trigger_init *trigs;
+	/** Number of triggers to be used */
+	int32_t nb_trigs;
+	/** IRQ descriptor to be used */
+	void *irq_desc;
```

Comment:
I am not 100% sure but I think all members of this struct up until irq_desc can be discarded, there is no reason to keep them in desc.

Reply:
Removed

Reply:
It could be that uart_desc is needed, you're right